File: VirtualAppDemo/app/src/main/java/com/jared/virtualappdemo/MainActivity.java
package com.jared.virtualappdemo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.jared.virtualappdemo.proxy.TestProxy;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.tv_start_activity).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, TestActivity.class);
                startActivity(intent);
            }
        });

        TestProxy.NoProxy();
        TestProxy.StaticProxy();
        TestProxy.DynamicProxy();
    }
}

File: DexPack/ApkTest/app/src/main/java/com/jared/apktest/MainActivity.kt
package com.jared.apktest

import android.content.Intent
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_test.*

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d("MainActivity", "onCreate")
        setContentView(R.layout.activity_test)
        tvName.text = showStr()
        btNext.setOnClickListener {
            gotoSecondActivity()
        }
    }

    private fun showStr(): String {
        return "Hello，I am ApkTest!"
    }

    private fun gotoSecondActivity() {
        startActivity(Intent(this, SecondActivity::class.java))
    }
}
